### PR TITLE
Feature/enhancements

### PR DESCRIPTION
**Pull request by aarti-joshi-nttd**
This PR attempts to enhance the application gateway module.

1. Add variables to the module so more configuration options are available.
2. Provide ability to create private Application gateway
3. Rename folders/variables appropriately
4. Modify the tests and make sure test runs are successful.

### CODE REVIEW

File: README.md
Comment (chris11-taylor-nttd) [inline README.md:1]: This file needs the heading and overview updated.

File: examples/private_app_gateway/main.tf [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  source = "git::https://github.com/launchbynttdata/tf-azurerm-module_primitive-nsg_subnet_association.git?ref=1.0.0"
+
+  network_security_group_id = module.network_security_group.network_security_group_id
+  subnet_id                 = local.app_gtwy_subnet_id
+
+  depends_on = [module.network_security_group]
+
+}
+
+module "app-gateway" {
```
Comment (chris11-taylor-nttd) [inline examples/private_app_gateway/main.tf]: Any reason this is `kebab-case` where typically the module references we use are `snake_case`?

File: examples/with_web_app/test.tfvars [hunk 2]
```diff
@@ -18,33 +18,29 @@ app_gateways = {
...
+      name               = "web_app_http_listener"
       frontend_port_name = "frontend_port"
       protocol           = "Http"
     }]
     client_name           = "launch"
     environment           = "demo"
-    location              = "East Us 2"
-    location_short        = "eastus2"
+    location              = "East Us2"
+    location_short        = "eastus"
```
Comment (chris11-taylor-nttd) [inline examples/with_web_app/test.tfvars]: This `location_short` doesn't match the `location`, and I see a few other references to `eastus` further on in this file. Should these resources be limited to a single region or is the intent that they span regions?

**Review by debasish-sahoo-nttd — APPROVED** [review]

File: examples/private_app_gateway/main.tf (hunk 1 above)
Comment (aarti-joshi-nttd) [reply examples/private_app_gateway/main.tf]: Fixed this!!

File: examples/with_web_app/test.tfvars (hunk 2 above)
Comment (aarti-joshi-nttd) [reply examples/with_web_app/test.tfvars]: Fixed it!!

File: README.md
Comment (aarti-joshi-nttd) [reply README.md:1]: Updated and added some notes I collected during testing

**Review by chris11-taylor-nttd — APPROVED** [review]

**Pull request merged**